services/core-web/src/components/noticeOfWork/applications/ViewNoticeOfWorkApplication.js

```diff
@@ -43,21 +49,29 @@ const propTypes = {
 };
 
 export class ViewNoticeOfWorkApplication extends Component {
-  state = {
-    isLoaded: false,
-    showOriginalValues: false,
-  };
+  static noticeOfWorkPageFromRoute = {};
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isLoaded: false,
+      showOriginalValues: false,
+    };
+  }
 
   componentDidMount() {
     const { id } = this.props.match.params;
     this.props.fetchNoticeOFWorkActivityTypeOptions();
     this.props.fetchImportedNoticeOfWorkApplication(id).then(() => {
-      this.props.history.push(
-        routes.VIEW_NOTICE_OF_WORK_APPLICATION.hashRoute(id, "#application-info")
-      );
       this.setState({ isLoaded: true });
     });
     this.props.fetchOriginalNoticeOfWorkApplication(id);
+    this.noticeOfWorkPageFromRoute =
+      this.props.location &&
+      this.props.location.state &&
+      this.props.location.state.noticeOfWorkPageFromRoute
+        ? this.props.location.state.noticeOfWorkPageFromRoute
+        : this.noticeOfWorkPageFromRoute;
```

Comment:
So I don't think this variable `noticeOfWorkPageFromRoute` is needed, you should be able to use`this.props.location.state.noticeOfWorkPageFromRoute` as your check

Reply:
I created the variable `noticeOfWorkPageFromRoute` because I need to access the value of `this.props.location.state.noticeOfWorkPageFromRoute` in a few other places, so instead of doing the long null check in those other places as well, I stored the result of it once.
I have also switched it from being a static variable to a state variable now.